modules/ballerina-core/src/main/java/org/ballerinalang/model/values/BMessage.java

```diff
@@ -249,6 +255,17 @@ public BMessage clone() {
                 this.value.getMessageDataSource() instanceof BallerinaMessageDataSource) {
             clonedMessage.setMessageDataSource(((BallerinaMessageDataSource) this.value.
                     getMessageDataSource()).clone());
+            isDataAdded = true;
+        }
+
+        // This code block is added to fix the bug related to workers when accessing message variable within
+        // resource. Fix #2623
+        if (!isDataAdded) {
+            try {
+                Thread.sleep(1);
```

Comment:
When workers are started within a resource, the message variable is accessible within those workers. By the time worker trying to access the content of the message, there is a chance that data is still not filled into the buffers within the message (since IO operations are running on a separate netty worker thread). Due to this reason, even though we clone the message and pass that to the worker, actual data is not available by the time we make the clone. Which means that workers will get messages with empty content.
Due to this reason, we are waiting here until the message content is added to the message through netty worker thread.